### ISSUE TITLE
Analyzer reports 3 issues for 1 problem involving a non-const arg to a const constructor in a list

### ISSUE THREAD

**Issue by srawlins** [dart-lang/sdk#53743]
Given this code:

```dart

void f(int y) {
  const list = [
    C(y),
    C(),
    ...[C(), C(), C()],
  ];
  print(list);
}

class C {
  const C([int? x = 1]);
}
```

Analyzer produces 3 diagnostics on the `y` argument in `C(y)`:

* Arguments of a constant creation must be constant expressions.[ (view docs)](https://dart.dev/diagnostics/const_with_non_constant_argument)
Try making the argument a valid constant, or use 'new' to call the constructor.
* The values in a const list literal must be constants.[ (view docs)](https://dart.dev/diagnostics/non_constant_list_element)
Try removing the keyword 'const' from the list literal.
* Const variables must be initialized with a constant value.[ (view docs)](https://dart.dev/diagnostics/const_initialized_with_non_constant_value)
Try changing the initializer to be a constant expression.

Two issues:

* These should be compressed into one error. I think the first is a fair one to keep, over the others.
* The suggestion in the first, "use 'new'" is not helpful, unnecessary, and erroneous.

I think the problems can be solved by considering whether a const constructor call is _explicit_ (with `const`) or implicitly const because of a const context.

**Comment by bwilkerson** [comment]
I agree that the first is the root cause. It looks like we're deciding that an instance creation expression isn't a const expression if some of the arguments aren't const expressions, but I think we should recover by assuming that it _will be_ a constant expression when the deeper issue is addressed. I think that doing so would remove the last two diagnostics naturally.
